README.md
# EasyToast

![Swift 5.0](https://img.shields.io/badge/Swift-5.0-blue.svg)
[![Version](https://img.shields.io/cocoapods/v/EasyToast.svg?style=flat)](http://cocoapods.org/pods/EasyToast)
[![License](https://img.shields.io/cocoapods/l/EasyToast.svg?style=flat)](http://cocoapods.org/pods/EasyToast)
[![Platform](https://img.shields.io/cocoapods/p/EasyToast.svg?style=flat)](http://cocoapods.org/pods/EasyToast)

Android-like toast with simple interface, using a toast queue to handle multiple toasts allowing to push or present a ViewController without disappear 

## Requirements
- Swift: 4.x
- iOS: 8+

## Swift Version
Depending on which version of Swift your project uses, you should use specific versions of EasyToast.

|Swift version         |EasyToast version   |
|:---------------------|:-------------------|
|Swift 4.2 - 5.0       |`3.0.0` or later    |
|Swift 3               |`2.0.x`             |
|Swift 2.2             |`1.1.2`             |


## Example Project

To run the example project, clone the repo, and run `pod install` from the Example directory first.

## Installation

EasyToast is available throgugh [CocoaPods](http://cocoapods.org). To install
it, simply add the following line to your Podfile:

```ruby
pod "EasyToast"
```

## Show Toasts

### Show default toasts

```swift
self.view.showToast("Toast Text", position: .Bottom, popTime: 5, dismissOnTap: false)
self.view.showToast("Toast Text", position: .Bottom, popTime: kToastNoPopTime, dismissOnTap: true)
```

### Personalize toasts

```swift
self.view.showToast("Toast Text", position: .Bottom, popTime: 5, dismissOnTap: false, bgColor: UIColor.blackColor(), textColor: UIColor.whiteColor(), font: UIFont.boldSystemFontOfSize(19))
self.view.showToast("Toast Text", position: .Bottom, popTime: kToastNoPopTime, dismissOnTap: true, bgColor: UIColor.redColor(), textColor: UIColor.blackColor(), font: UIFont.boldSystemFontOfSize(19))
```

### Personalize all toasts in view

```swift
self.view.toastBackgroundColor = UIColor.blackColor().colorWithAlphaComponent(0.7)
self.view.toastTextColor = UIColor.redColor()
self.view.toastFont = UIFont.boldSystemFontOfSize(19)

self.view.showToast("Toast Text", position: .Bottom, popTime: 5, dismissOnTap: false)
self.view.showToast("Toast Text", position: .Bottom, popTime: kToastNoPopTime, dismissOnTap: true)
```

## Example
On button click will be executed this code

```swift
self.view.toastBackgroundColor = UIColor.blackColor().colorWithAlphaComponent(0.7)
self.view.toastTextColor = UIColor.whiteColor()
self.view.toastFont = UIFont.boldSystemFontOfSize(19)
        
self.view.showToast("Toast", position: .Bottom, popTime: 2, dismissOnTap: false)
self.view.showToast("Dismiss on tap toast", position: .Bottom, popTime: kToastNoPopTime, dismissOnTap: true, bgColor: UIColor.redColor().colorWithAlphaComponent(0.7), textColor: UIColor.whiteColor(), font: UIFont.systemFontOfSize(19))
self.view.showToast("Long Text Toast:\n" +
    "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas vitae elit non leo pellentesque feugiat. Cras nec volutpat massa, nec blandit nisi. Etiam ut hendrerit purus. Morbi accumsan, risus ut cursus finibus, quam ipsum egestas nisl, vel hendrerit massa justo nec metus. Cras pulvinar, leo eu pulvinar convallis, tellus felis laoreet massa, ac tincidunt orci massa non odio. Ut pulvinar dictum metus quis mollis. Aenean tincidunt sit amet turpis sed egestas. Morbi porta dolor neque, ut pellentesque urna semper id. Cras rhoncus consequat justo. Cras dictum enim orci, ac vestibulum enim cursus id.",
    position: .Bottom,
    popTime: 5,
    dismissOnTap: true,
    bgColor: UIColor.redColor().colorWithAlphaComponent(0.7),
    textColor: UIColor.whiteColor(),
    font: UIFont.systemFontOfSize(19))
```

![EasyToast Gif Example](https://raw.github.com/f-meloni/EasyToast/master/GifExample/EasyToastExample.gif)

## Configure Toasts

```swift
EasyToastConfiguration.toastInnerPadding = 10
EasyToastConfiguration.animationDuration = 0.6
EasyToastConfiguration.initialSpringVelocity = 0.07
EasyToastConfiguration.dampingRatio = 0.65
```

## Toast tag

To avoid multiple toasts with the same argument, e.g. if a user taps many times on the same button that shows a toast, you can tag it to prevent multiple toast displaying

#### Example

```swift
view.showToast("Toast", tag:"test", position: .Bottom, popTime: kToastNoPopTime, dismissOnTap: false)
view.showToast("Toast", tag:"test", position: .Bottom, popTime: kToastNoPopTime, dismissOnTap: false)
view.showToast("Toast", tag:"test", position: .Bottom, popTime: kToastNoPopTime, dismissOnTap: false)
```

This code will show only the first toast

## Screenshots

![EasyToast Screenshot](https://raw.github.com/f-meloni/EasyToast/master/Screenshots/EasyToastScreenshots.jpg)

## Author

Franco Meloni, franco.meloni91@gmail.com

## License

EasyToast is available under the MIT license. See the LICENSE file for more info.


CHANGELOG.md
#CHANGELOG

##Version 3.0.0
Swift 5 support

##Version 2.0.2
Fixes on toast displaying, changed iphone toast max size"

##Version 2.0.1
Minor fixes

##Version 2.0.0
Swift 3 support

##Version 1.1.2
Fix: Present a toast now doesn't make your status bar appear if is hidden

##Version 1.1.1
Added toasts configuration

##Version 1.0.1
Toast popTime now is a Double

##Version 1.0.0
Added toasts tag, bugfix

##Version 0.9.3
Added changelog

##Version 0.9.2
Added more documentation

##Version 0.9.1
Minor fixes

##Version 0.9
Fix: Status bar color now doesn't change when a toast is presented


Example/EasyToast/ViewController.swift
//
//  ViewController.swift
//  EasyToast
//
//  Created by Franco Meloni on 08/05/2016.
//  Copyright (c) 2016 Franco Meloni. All rights reserved.
//

import UIKit
import EasyToast

class ViewController: UIViewController {
    fileprivate lazy var button : UIButton = {
        let button = UIButton(type: .system)
        
        button.frame = CGRect(x: 0, y: 50, width: self.view.frame.width, height: 50)
        button.titleLabel?.font = UIFont.systemFont(ofSize: 21)
        button.setTitle("Send Toasts", for: UIControl.State())
        button.addTarget(self, action: #selector(showToast), for: .touchUpInside)
        
        return button
    }()
    
    override func viewDidLoad() {
        super.viewDidLoad()

        self.view.addSubview(self.button)
        
        EasyToastConfiguration.toastInnerPadding = 10
        EasyToastConfiguration.animationDuration = 0.6
        EasyToastConfiguration.initialSpringVelocity = 0.07
        EasyToastConfiguration.dampingRatio = 0.65
    }
    
    override func viewDidLayoutSubviews() {
        super.viewDidLayoutSubviews()
        
        self.button.frame = CGRect(x: 0, y: 50, width: self.view.frame.width, height: 50)
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    @objc func showToast() {
        self.view.toastBackgroundColor = UIColor.black.withAlphaComponent(0.8)
        self.view.toastTextColor = UIColor.white
        self.view.toastFont = UIFont.boldSystemFont(ofSize: 19)
        
        self.view.showToast("Toast", position: .bottom, popTime: 1.5, dismissOnTap: false)
        self.view.showToast("Dismiss on tap toast", position: .top, popTime: kToastNoPopTime, dismissOnTap: true, bgColor: UIColor.red.withAlphaComponent(0.7), textColor: UIColor.white, font: UIFont.systemFont(ofSize: 19))
        self.view.showToast("Long Text Toast:\n" +
            "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas vitae elit non leo pellentesque feugiat. Cras nec volutpat massa, nec blandit nisi. Etiam ut hendrerit purus. Morbi accumsan, risus ut cursus finibus, quam ipsum egestas nisl, vel hendrerit massa justo nec metus. Cras pulvinar, leo eu pulvinar convallis, tellus felis laoreet massa, ac tincidunt orci massa non odio. Ut pulvinar dictum metus quis mollis. Aenean tincidunt sit amet turpis sed egestas. Morbi porta dolor neque, ut pellentesque urna semper id.",
                            position: .bottom,
                            popTime: 5,
                            dismissOnTap: true,
                            bgColor: UIColor.red.withAlphaComponent(0.7),
                            textColor: UIColor.white,
                            font: UIFont.systemFont(ofSize: 19))
    }

    func closeModal() {
        self.dismiss(animated: true, completion: nil)
    }
}

